runtime/protocol/aws-protocol-core/common/src/aws/smithy/kotlin/runtime/awsprotocol/AwsQueryCompatibleErrorDetails.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.awsprotocol
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.ServiceErrorMetadata
+import aws.smithy.kotlin.runtime.ServiceException
+
+/**
+ * Header name for awsQuery-compatible error values.
+ */
+@InternalApi
+public const val X_AMZN_QUERY_ERROR_HEADER: String = "x-amzn-query-error"
```

Comment:
Can probably be `internal`

Reply:
Unfortunately must be public since the generated deserializer code is consuming it.